SFR/Assembly/outer_assembly/make_SFR_assembly_model.py

```diff
@@ -0,0 +1,209 @@
+am241.add_nuclide('Am241', 1.0)
+am241.set_density('g/cm3', 10.0)
+
+o16 = openmc.Material(name='O16')
+o16.add_nuclide('O16', 1.0)
+o16.set_density('g/cm3', 10.0)
+
+sodium = openmc.Material(name='Na')
+sodium.add_nuclide('Na23', 1.0)
+sodium.set_density('g/cm3', 0.96)
+
+cu63 = openmc.Material(name='Cu63')
+cu63.set_density('g/cm3', 10.0)
+cu63.add_nuclide('Cu63', 1.0)
+
+Al2O3 = openmc.Material(name='Al2O3')
+Al2O3.set_density('g/cm3', 10.0)
+Al2O3.add_element('O', 3.0)
+Al2O3.add_element('Al', 2.0)
+
+helium = openmc.Material(name='Helium for gap')
+helium.set_density('g/cm3', 0.001598)
+helium.add_element('He', 2.4044e-4)
+
+#Material mixtures
+
+outer_fuel_material= openmc.Material.mix_materials([u235, u238, pu238, pu239, pu240, pu241, pu242, am241, o16],[0.0018, 0.73, 0.0053, 0.0711, 0.0445, 0.0124, 0.0156, 0.0017, 0.1176],'wo')
+cladding_material = openmc.Material.mix_materials([cu63,Al2O3], [0.997,0.003], 'wo')
+
+materials=openmc.Materials([outer_fuel_material,sodium,helium,cladding_material])
+materials.export_to_xml()
+
+
+###############################################################################
+# Define problem geometry
+
+# Create cylindrical surfaces
+
+
+
+# Create cylindrical surfacesfuel_or = openmc.ZCylinder(surface_id=1, r=0.943/2) 
+
+fuel_or = openmc.ZCylinder(surface_id=1, r=0.943/2) 
+clad_ir = openmc.ZCylinder(surface_id=2, r=0.973/2) 
+clad_or = openmc.ZCylinder(surface_id=3, r=1.073/2) 
+
+
+
+z_co_ordinates=np.linspace(-height/2,height/2,N+1)
+z_plane=[]
+for i in range (len(z_co_ordinates)):
+    z_plane.append(openmc.ZPlane(z0=z_co_ordinates[i]))
+
+
+#set BCs
+z_plane[0].boundary_type='vacuum'
+z_plane[-1].boundary_type='vacuum'
+top=z_plane[-1]
+bottom=z_plane[0]
+
+outer_fuel_cells=[]
+outer_gas_gap_cells=[]
+outer_clad_cells=[]
+outer_sodium_cells=[]
+
+all_outer_cells=[]
+
+outer_fuel_cells=[]
+outer_gas_gap_cells=[]
+outer_clad_cells=[]
+outer_sodium_cells=[]
+
+for i in range (N):
+    
+    layer=+z_plane[i]&-z_plane[i+1]
+    
+    outer_fuel_cells.append(openmc.Cell( fill=outer_fuel_material,region=layer & -fuel_or))
+    outer_gas_gap_cells.append(openmc.Cell( fill=helium,region=layer & +fuel_or &-clad_ir))
+    outer_clad_cells.append(openmc.Cell(fill=cladding_material,region= layer& +clad_ir &-clad_or))
+    outer_sodium_cells.append(openmc.Cell(fill=sodium,region= +clad_or & layer))
+    
+    all_outer_cells.append(outer_fuel_cells[i])
+    all_outer_cells.append(outer_gas_gap_cells[i])
+    all_outer_cells.append(outer_clad_cells[i])
+    all_outer_cells.append(outer_sodium_cells[i])
+    
+    
+outer_u = openmc.Universe(cells=(all_outer_cells))
+
+sodium_mod_cell = openmc.Cell(cell_id=6, fill=sodium)
+sodium_mod_u = openmc.Universe(universe_id=3, cells=(sodium_mod_cell,))
+
+# Define a lattice for outer assemblies
+out_lat = openmc.HexLattice(lattice_id=1, name='outer assembly')
+out_lat.center = (0., 0.)
+out_lat.pitch = (21.08/17,)
+out_lat.orientation = 'y'
+out_lat.outer = sodium_mod_u
```

Comment:
consider using a 3D lattice, you can look at the GCMR repo to see an example of how to do this and still have density feedback